### PR TITLE
Faster Parser->literal()

### PR DESCRIPTION
**Pull request by staabm**
same as https://github.com/scssphp/scssphp/pull/657, but for 1.x

### CODE REVIEW

File: src/Parser.php [hunk 1]
```diff
@@ -2681,7 +2699,7 @@ protected function value(&$out)
 
         // unicode range with wildcards
         if (
-            $this->literal('U+', 2) &&
+            $this->literal(strtolower('U+'), 2) &&
```
Comment (stof) [inline src/Parser.php:2702]: why not writing it directly in lowercase ?

Comment (staabm) [reply src/Parser.php:2702]: I can do that, if you prefer. 

my reasoning was, that the actual notation in SCSS will have a uppercase U and like that we can easier find it in the codebase

**Comment by stof** [comment]
Do we actually need to lowercase the whole buffer which would double the memory usage of that parser ?

Maybe checking the current character against both `$what[0]` and `strtoupper($what[0])` could be enough to achieve most of the performance gain.

**Comment by staabm** [comment]
seems not worth further investigation

**Pull request closed without merging**